app/models/miq_report_result/result_set_operations.rb

```diff
@@ -41,10 +41,10 @@ def result_set_for_reporting(report_result, options)
       if result_set.present? && report
         if filtering_enabled?(options)
           result_set, count_of_full_result_set = report.filter_result_set(result_set, filter_options(options))
-          allowed_columns_to_format = report.col_order - filter_options(options).keys
+          allowed_columns_to_format = report.cols - filter_options(options).keys
         end
 
-        result_set.map! { |x| x.slice(*report.col_order) }
+        result_set.map! { |x| x.slice(*report.cols) }
```

Comment:
we want the columns that are in the col_order.
I suppose we may want the columns in cols as well.

Again, I'd prefer keeping both columns.

Not sure the proper code for this one

```suggestion
        result_set.map! { |x| x.slice(*(report.col_order + report.cols) }
```

Reply:
Ugh, need to handle the cases where col_order and cols are nil.
Think we need to slice the includes columns as well.

Reply:
this suggestion is not quite right

Reply:
maybe something like `(report.cols || [] ) | (report.col_order || [] )`

Reply:
I put it into method  `available_columns`

Reply:
we have method for it - `cols_for_report`